blocks/model.py

```diff
@@ -190,7 +179,13 @@ def get_params(self):
         parameters that do not belong to any brick.
 
         """
-        return self.params
+        if (len(set(parameter.name for parameter in self.parameters)) !=
+                len(self.parameters)):
+            raise ValueError("parameters should have different names")
```

Comment:
@mohammadpz , names of parameters must be different for one brick. Likewise names of children of a brick and names of top bricks must be different. This line ensures that the users and Blocks developers take care of  it...